csrc/scheduler/registry.cpp

```diff
@@ -1999,6 +1999,8 @@ class PersistentKernelScheduler : public SchedulerEntry {
       }
     }
 
+    TORCH_INTERNAL_ASSERT(is_cross_grid && cross_grid_params.has_value())
```

Comment:
I don't think this is correct. This should be:

```
if (is_cross_grid) {
   TORCH_INTERNAL_ASSERT(cross_grid_params.has_value());
}
```

Reply:
How about this? 
`TORCH_INTERNAL_ASSERT(!is_cross_grid || cross_grid_params.has_value())`

Reply:
Yes, that should be fine too.